tests/output/test_io_mesh.py

```diff
@@ -0,0 +1,45 @@
+from firedrake import *
+import pytest
+from petsc4py import PETSc
+from petsc4py.PETSc import ViewerHDF5
+from pyop2.mpi import COMM_WORLD
+import os
+
+
+@pytest.mark.parallel(nprocs=4)
+@pytest.mark.parametrize('outformat', [ViewerHDF5.Format.HDF5_XDMF,
+                                       ViewerHDF5.Format.HDF5_PETSC])
+@pytest.mark.parametrize('heterogeneous', [False, True])
+def test_io_mesh(outformat, heterogeneous, tmpdir):
+    # Parameters
+    fname = os.path.join(str(tmpdir), "test_io_mesh_dump.h5")
+    fname = COMM_WORLD.bcast(fname, root=0)
+    ntimes = 3
+    # Create mesh.
+    mesh = RectangleMesh(4, 1, 4., 1.)
+    mesh.init()
```

Comment:
It turned out that it would just work as adding partition overlaps does not change what we save; tested just in case. So I've removed this `mesh.init()`.